packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManager.java

```diff
@@ -661,6 +661,10 @@ public static long measureText(
       for (int lineIndex = 0; lineIndex < calculatedLineCount; lineIndex++) {
         boolean endsWithNewLine =
             text.length() > 0 && text.charAt(layout.getLineEnd(lineIndex) - 1) == '\n';
+        if (!endsWithNewLine && lineIndex + 1 < layout.getLineCount()) {
+          calculatedWidth = width;
+          break;
+        }
```

Comment:
I think we may also reach this path when `widthYogaMeasureMode` is `YogaMeasureMode.UNDEFINED`, in which case we are asking for max-content size, and the `width` is `NaN`.

Reply:
I think on L604, we could replace:

```java
if (widthYogaMeasureMode == YogaMeasureMode.AT_MOST && calculatedWidth > width)
```

with

```java
if (widthYogaMeasureMode == YogaMeasureMode.AT_MOST && (calculatedWidth > width || layout.getLineCount() > 1))
```